File: packages/api/app/schemas/subscription.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class SubscriptionResponse(BaseModel):
    id: str
    user_id: str
    plan: str
    credits_used: int
    images_used: int
    credits_remaining: int  # computed: limit - used (-1 = unlimited)
    images_remaining: int   # computed
    is_active: bool

    model_config = {"from_attributes": True}


class PlanInfo(BaseModel):
    plan: str
    price: int
    credits_per_month: int  # -1 = unlimited
    images_per_month: int
    max_projects: int


class SubscriptionUpgradeRequest(BaseModel):
    plan: str  # "starter" | "pro" | "enterprise"
    payment_method_id: Optional[str] = None  # Stripe PM ID


class AdminUserItem(BaseModel):
    id: str
    email: str
    full_name: str
    plan: str
    created_at: datetime


class AdminStatsResponse(BaseModel):
    total_users: int
    active_subscriptions: int
    pro_users: int
    starter_users: int
    free_users: int
    total_projects: int
    total_estimates: int
    # AI 비용 통계
    ai_calls_today: int = 0
    ai_calls_this_month: int = 0
    ai_cost_this_month_usd: float = 0.0
    ai_cost_this_month_krw: int = 0  # USD × 1350 환율
    failed_ai_requests: int = 0


class AdminAIModelBreakdown(BaseModel):
    model: str
    call_count: int
    total_cost_usd: float
    total_input_tokens: int
    total_output_tokens: int


class AdminAIStatsDetail(BaseModel):
    daily_calls: list[dict]          # [{date, count, cost_usd}]
    model_breakdown: list[AdminAIModelBreakdown]
    top_users: list[dict]            # [{user_id, email, credits_used, cost_usd}]

File: packages/web/types/subscription.ts
export interface Subscription {
  id: string
  user_id: string
  plan: 'free' | 'starter' | 'pro' | 'enterprise'
  credits_used: number
  images_used: number
  credits_remaining: number // -1 = unlimited
  images_remaining: number
  is_active: boolean
}

export interface PlanInfo {
  plan: string
  price: number
  credits_per_month: number // -1 = unlimited
  images_per_month: number
  max_projects: number
}

export interface AdminStats {
  total_users: number
  active_subscriptions: number
  pro_users: number
  starter_users: number
  free_users: number
  total_projects: number
  total_estimates: number
  // AI 비용 통계
  ai_calls_today?: number
  ai_calls_this_month?: number
  ai_cost_this_month_usd?: number
  ai_cost_this_month_krw?: number
}

export interface ModelBreakdown {
  model: string
  call_count: number
  total_cost_usd: number
  total_input_tokens: number
  total_output_tokens: number
}

export interface AIStatsDetail {
  daily_calls: { date: string; count: number; cost_usd: number }[]
  model_breakdown: ModelBreakdown[]
  top_users: { user_id: string; email: string; credits_used: number; cost_usd: number }[]
}

export interface AdminUser {
  id: string
  email: string
  full_name: string
  plan: string
  created_at: string
}
